### PR TITLE
Set up group page infrastructure (actions, components, controllers, & stores)

### LINKED ISSUE #571: Create group page component

### LINKED ISSUE #572: Create group grid (don't worry too much about styling within it for now!)

### LINKED ISSUE #573: Create group store

### LINKED ISSUE #574: Create group actions

### LINKED ISSUE #575: Create non-api groups controller

### LINKED ISSUE #578: Create group card and render already-made students card components

### LINKED ISSUE #594: Scope api groups controller under api conferences controller

### LINKED ISSUE #595: Scope non-api groups controller under non-api conferences controller
Need to touch `routes.rb` and `groups_controller.rb` (need to move `groups_controller into a folder called `conferences`!

Remove the `index` action in `groups_controller.rb` as well.

### LINKED ISSUE #620: Render groups belonging to conference on conference page

### LINKED ISSUE #621: Finish group page infrastructure
Finish @sonia-y's pull!

### PR DESCRIPTION
**Pull request by sonia-y**
Fix #571 
Fix #572 
Fix #573 
Fix #574 
Fix #575 
Fix #578
Fix #594 
Fix #595 
Fix #620
Fix #621

### CODE REVIEW

File: app/assets/javascripts/components/group/group_page.es6.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        paddingRight: '196px',
+      },
+    };
+  }
+
+  // --------------------------------------------------
+  // Lifecycle
+  // --------------------------------------------------
+  componentWillMount() {
+  	this.setState(ProfileStore.getState());
```
Comment (warrenshen) [inline app/assets/javascripts/components/group/group_page.es6.jsx]: Spacing messed up!

File: app/assets/javascripts/components/group/group_page.es6.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  	ProfileActions.fetchProfile();
+  }
+
+  componentWillUnmount() {
+    ProfileStore.unlisten(this._listener);
+  }
+
+  render() {
+    return (
+        <div style={this.styles.container}>
```
Comment (warrenshen) [inline app/assets/javascripts/components/group/group_page.es6.jsx]: Spacing messed up!

File: app/assets/javascripts/components/group/group_page.es6.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          <Sidebar
+            hidden={false}
+            profile={this.state.profile} />
+          <GroupGrid  />
```
Comment (warrenshen) [inline app/assets/javascripts/components/group/group_page.es6.jsx]: There's an extra space after `GroupGrid`?

File: app/assets/javascripts/components/user/student_outreach.es6.jsx [hunk 4]
```diff
@@ -34,14 +34,15 @@ class StudentOutreach extends Component {
...
     var responsibility = this.props.student.responsibility;
-    if (responsibility) {
-      return (
-        <Clickable
-          content={'Volunteer'}
-          route={RouteConstants.users.show(responsibility.user.id)}
-          type={'h6'} />
-      );
-    }
+    // TODO(Warren & Unzi): Fix this with responsibility serializers.
```
Comment (warrenshen) [inline app/assets/javascripts/components/user/student_outreach.es6.jsx]: If you merge master now this should be fixed and you can uncomment this!

File: app/assets/javascripts/stores/group_store.es6.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  class GroupStore {
+
+    constructor() {
+      this.overlay = {
+        active: false,
+        target: '',
+        type: '',
+      };
+      this.sidebar = true;
+      this.group = {
```
Comment (warrenshen) [inline app/assets/javascripts/stores/group_store.es6.jsx]: Alphabetize!

File: app/assets/javascripts/actions/group_actions.es6.jsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    constructor() {
+      this.generateActions(
+        'storeGroup',
+        'toggleSidebar'
+      );
+    }
+
+    fetchGroup(conferenceId, id) {
+      console.log(conferenceId + ' ' + id);
+      resolve = (response) => this.storeGroup(response);
```
Comment (warrenshen) [inline app/assets/javascripts/actions/group_actions.es6.jsx]: `var resolve` and `console.log`!

File: app/controllers/conferences/groups_controller.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class Conferences::GroupsController < BaseController
+
+  def index
+  end
+
+  def show
+  	@conferenceId = params[:conference_id].to_i
```
Comment (warrenshen) [inline app/controllers/conferences/groups_controller.rb]: Spacing!

File: app/views/conferences/groups/show.html.erb [hunk 8]
```diff
@@ -0,0 +1 @@
+<%= react_component('GroupPage', { conferenceId: @conferenceId, id: @id }) %>
```
Comment (warrenshen) [inline app/views/conferences/groups/show.html.erb]: We can pass this props down as `id`.

File: config/routes.rb [hunk 9]
```diff
@@ -8,12 +8,19 @@
 
   get 'profile', to: 'users#profile'
 
-  resources :conferences, only: [:index, :show]
+
+
```
Comment (warrenshen) [inline config/routes.rb]: Extra spacing!

**Comment by warrenshen** [comment]
@maxwolffe I want you to jump on this branch and finish it up (reassigned some of @sonia-y's issues to you).

**Comment by warrenshen** [comment]
@maxwolffe make sure to merge master before starting, you're probably going to get hit by a few merge conflicts!

File: app/views/conferences/groups/show.html.erb (hunk 8 above)
Comment (maxwolffe) [inline app/views/conferences/groups/show.html.erb]: What do you mean by this @warrenshen? We shouldn't need the conferenceId here at all right?

Comment (warrenshen) [inline app/views/conferences/groups/show.html.erb]: We can just call it `id`!

File: app/assets/javascripts/components/group/group_grid.es6.jsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    return {
+      group: React.PropTypes.object.isRequired,
+    };
+  }
+
+  static get defaultProps() {
+    return {
+      group: {},
+    };
+  }
```
Comment (warrenshen) [inline app/assets/javascripts/components/group/group_grid.es6.jsx]: Don't need default props!

File: app/assets/javascripts/components/group/group_page.es6.jsx [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+      id: React.PropTypes.number.isRequired,
+    };
+  }
+
+  static get defaultProps() {
+    return {
+    	conferenceId: 1,
+      id: 1,
+    };
+  }
```
Comment (warrenshen) [inline app/assets/javascripts/components/group/group_page.es6.jsx]: Don't need default props!

File: app/assets/javascripts/components/group/group_page.es6.jsx [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+    super(props);
+    this._listener = (state) => this.setState(state);
+  }
+
+  // --------------------------------------------------
+  // Props
+  // --------------------------------------------------
+  static get propTypes() {
+    return {
+    	conferenceId: React.PropTypes.number.isRequired,
```
Comment (warrenshen) [inline app/assets/javascripts/components/group/group_page.es6.jsx]: Spacing!

File: app/assets/javascripts/constants/api_constants.es6.jsx [hunk 13]
```diff
@@ -3,6 +3,14 @@
 
     get conferences() {
       return {
+        groups: {
+          show: function(conferenceId, id) {
+            return `/api/conferences/${conferenceId}/groups/${id}`
+          },
+          update: function(conferenceId, id) {
+            return `/api/conferences/${conferenceId}/groups/${id}`
+          },
```
Comment (warrenshen) [inline app/assets/javascripts/constants/api_constants.es6.jsx]: We can use arrow syntax here!

File: app/assets/javascripts/stores/group_store.es6.jsx [hunk 14]
```diff
@@ -0,0 +1,24 @@
+(() => {
+  class GroupStore {
+
+    constructor() {
+      this.sidebar = true;
```
Comment (warrenshen) [inline app/assets/javascripts/stores/group_store.es6.jsx]: Alphabetize!

File: app/controllers/conferences/groups_controller.rb (hunk 7 above)
Comment (warrenshen) [inline app/controllers/conferences/groups_controller.rb]: Spacing!

File: app/assets/javascripts/components/group/group_page.es6.jsx [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+      },
+    };
+  }
+
+  // --------------------------------------------------
+  // Lifecycle
+  // --------------------------------------------------
+  componentWillMount() {
+  	this.setState(ProfileStore.getState());
+  	this.setState(GroupStore.getState());
```
Comment (warrenshen) [inline app/assets/javascripts/components/group/group_page.es6.jsx]: Spacing!

File: app/assets/javascripts/components/group/group_page.es6.jsx [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  componentDidMount() {
+  	ProfileStore.listen(this._listener);
+  	GroupStore.listen(this._listener);
+  	ProfileActions.fetchProfile();
+  	GroupActions.fetchGroup(this.props.conferenceId, this.props.id)
```
Comment (warrenshen) [inline app/assets/javascripts/components/group/group_page.es6.jsx]: Spacing!

**Pull request merged**